### PR TITLE
Add request password UI on backup export

### PR DESCRIPTION
**Pull request by mikeger**
- Includes the copy update and visual fixes for the back up screen.
- Also includes the tests.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Settings/BackupPasswordViewController.swift [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        navigationItem.rightBarButtonItem?.tintColor = .accent()
+        navigationItem.rightBarButtonItem?.isEnabled = false
+    }
+    
+    fileprivate func updateState(with text: String) {
+        password = Password(text)
+        navigationItem.rightBarButtonItem?.isEnabled = nil != password
+    }
+    
+    dynamic fileprivate func cancel() {
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Settings/BackupPasswordViewController.swift]: We can already start by doing `@objc private func` to be ready for Swift 4.

File: Wire-iOS/Sources/UserInterface/Settings/BackupPasswordViewController.swift [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
+     
+        if let _ = string.rangeOfCharacter(from: CharacterSet.newlines) {
+            if let _ = password {
+                completeWithCurrentResult()
+            }
+            return false
+        }
+        
+        let newString = ((textField.text ?? "") as NSString).replacingCharacters(in: range, with: string)
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Settings/BackupPasswordViewController.swift]: I guess we should also think about whitespaces (I assume a password only consisting of spaces is not allowed?).

Comment (mikeger) [reply Wire-iOS/Sources/UserInterface/Settings/BackupPasswordViewController.swift]: No, only whitespaces password is not good.

File: Wire-iOS/Sources/UserInterface/Settings/BackupPasswordViewController.swift [hunk 3]
```diff
@@ -24,15 +24,15 @@ struct Password {
     let value: String
     
     init?(_ value: String) {
-        guard value.count >= Password.minimumCharacters else { return nil }
+        guard value.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines).count >= Password.minimumCharacters else { return nil }
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Settings/BackupPasswordViewController.swift]: Maybe we should also forbid the user to enter whitespaces altogether

Comment (mikeger) [reply Wire-iOS/Sources/UserInterface/Settings/BackupPasswordViewController.swift]: I was advised to make it work exactly as the other password field in registration, there entering spaces is not forbidden.

**Pull request merged**